scipy/signal/tests/test_signaltools.py

```diff
@@ -1115,7 +1106,7 @@ def test_invalid_dtypes(self, dtype):
     def test_none(self):
         # gh-1651, trac #1124. Ensure this does not segfault.
         with pytest.warns(UserWarning):
-            assert_raises(TypeError, signal.medfilt, None)
+            assert_raises(ValueError, signal.medfilt, None)
         # Expand on this test to avoid a regression with possible contiguous
         # numpy arrays that have odd strides. The stride value below gets
         # us into wrong memory if used (but it does not need to be used)
```

Comment:
This is not the fault of this PR (actually, since the introduction of this hunk in https://github.com/scipy/scipy/commit/ce7e474b3df009c36a9710276d98875eecb23b2c), but this second part of the test has nothing to do with `test_none`, and should just get a separate `def test_odd_strides` (or whatever other name).